report/src/main/webapp/app/components/generatePdf.directive.js

```diff
@@ -0,0 +1,432 @@
+/*
+ * AET
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(['angularAMD', 'blob-stream', 'canvas2pdf', 'metadataService'], function (angularAMD, blobStream, canvas2pdf) {
+  'use strict';
+  angularAMD.directive('aetGeneratePdfReport', ['metadataService', generatePDFDirective]);
+
+  function generatePDFDirective(metadataService) {
+
+    return {
+      restrict: 'A',
+      link: linkFunc
+    };
+
+    function linkFunc(scope) {
```

Comment:
I will see what I can do about the size of that file.

Reply:
About other stuff:
- Stats will be taken from the metadata, that's my bad I haven't though of that before.
- I don't know how right now a user would load their own plugin. I couldn't use any good library myself so I had to write all the generating and drawing because of that. Maybe when the code gets updated then we can think about some advanced and good-looking charts. 
- I will take a look at these repetitions again and maybe figure out something cleaner.

Thanks!

Reply:
@wiiitek 
I've created services for both data part and drawing part of the report, so now instead of one big file we've got three smaller ones. The functionality stays unchanged though. 

Please tell me what you think about that approach.